### PR TITLE
Parameterize metadata address, listen port

### PR DESCRIPTION
**Pull request by LLParse**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,6 +23,18 @@ func main() {
...
 	app.Usage = "An external resource based scheduler for Rancher."
 	app.Action = run
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "metadata-address",
+			Usage: "The metadata service address",
+			Value: "rancher-metadata",
+		},
+		cli.StringFlag{
+			Name:  "listen",
```
Comment (alena1108) [inline main.go]: need a second opinion on this flag naming. @cjellick let us know if you are ok with "listen" or should it be more descriptive, something like "healtcheck-port"?

**Comment by cjellick** [comment]
Why are we doing this?

File: main.go (hunk 1 above)
Comment (LLParse) [reply main.go]: I noticed we used the term `listen` elsewhere, so I followed it. `port` would be more descriptive.

**Comment by LLParse** [comment]
@cjellick For environments where metadata isn't running on port 80, or DNS isn't available, or some combination of the two.

File: main.go [hunk 2]
```diff
@@ -25,6 +23,18 @@ func main() {
...
+			Usage: "Listen on this port for healthchecks",
+			Value: ":80",
```
Comment (cjellick) [inline main.go]: Having to include the `:` is bad. Don't want to do that

File: main.go (hunk 1 above)
Comment (cjellick) [reply main.go]: I'd prefer health-check port. No need to continue the bad convention :-)

File: main.go (hunk 2 above)
Comment (LLParse) [reply main.go]: I did this so we could listen on a specific interface, instead of all interfaces. Just my $.02, makes the param more flexible.

Comment (cjellick) [reply main.go]: But the usage doesnt indicate that you can specify a different interface and the log message has 0.0.0.0 in it.
...And from a practical standpoint, I don't think the scheduler is ever going to have more than one interface to listen for healtchecks on.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**